cv32e40x/tests/programs/custom/pma/pma.c

```diff
@@ -31,8 +31,50 @@
 static volatile uint32_t mcause = 0;
 static volatile uint32_t mepc = 0;
 static volatile uint32_t mtval = 0;
+static volatile uint32_t retpc = 0;
 
-static void (*cause_instr_access_fault)(void) = (void (*)(void))IO_ADDR;
+// Exception-causing instructions
+static void (*instr_access_fault)(void) = (void (*)(void))IO_ADDR;
+void misaligned_store(void) {
+  uint32_t tmp;
+  tmp = 0xBBBBBBBB;
+  __asm__ volatile("sw %0, 1(%1)" : "=r"(tmp) : "r"(IO_ADDR));
+}
+void load_misaligned_io(void) {__asm__ volatile("lw t0, 3(%0)" : : "r"(IO_ADDR));}
+void load_misaligned_iomem(void) {__asm__ volatile("lw t0, 0(%0)" : : "r"(MEM_ADDR_1 - 3));}
+void load_misaligned_memio(void) {__asm__ volatile("lw t0, 0(%0)" : : "r"(IO_ADDR - 1));}
+void store_first_access(void) {__asm__ volatile("sw %0, 2(%1)" : : "r"(0x11223344), "r"(IO_ADDR));}
```

Comment:
Admittedly, it is not very clean that these parts are defined so far away from the context where they are used. These magic numbers make sense within that context.
If anyone can conjure a better way to solve this problem then I would gladly follow that route instead.